build-python.sh

```diff
@@ -352,26 +356,26 @@ if test "${BUILD_WHEEL}" = true; then
     python -m build \
         --wheel \
         --outdir ../dist \
-        ${BUILD_ARGS} \
+        ${BUILD_ARGS:+${BUILD_ARGS}} \
```

Comment:
Please, explain this. What specifically is `shellcheck` complaining about that led to this change?

Reply:
Thanks, but I just pushed commits reverting this for the reasons mentioned in https://github.com/microsoft/LightGBM/pull/6641#discussion_r1804007090.

The claim you've made here that this being empty would result in the command failing is simply not true. You can try this yourself:

```shell
sh build-python.sh bdist_wheel
```

That will reach this line with `BUILD_ARGS` empty, and nothing will fail. We even intentionally set `BUILD_ARGS` to the empty string when `--precompile` is passed.

https://github.com/microsoft/LightGBM/blob/668bf5dadf1eb9a846302b2b76a313fbbef52870/build-python.sh#L295-L299

And I use `sh build-python.sh install --precompile` every time I develop on LightGBM's Python package.